docs/development/building-from-sources.md

```diff
@@ -2,85 +2,119 @@
 
 # Building from sources
 
-Building is easiest on Linux and relatively straightforward on Mac. For Windows,
-we currently recommend using the Docker image (described below) to build
-Pyodide. Another option for building on Windows is to use
+```{warning}
+If you are building the latest development version of Pyodide from the `main`
+branch, please make sure you are also following build instructions from the dev
+version of the documentation at
+[pyodide.org/en/latest/](https://pyodide.org/en/latest/development/building-from-sources.html)
+```
+
+Building is easiest with Docker. You can also build on Linux with `make`, while
+builing on Mac is possible, but there are known issues as of 0.18. For
+Windows, we currently recommend using Docker to build Pyodide. Another option
+for building on Windows is to use
 [WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10) to create a
 Linux build environment.
 
-## Build using `make`
+## Build instructions
+
+### Using Docker
+
+We provide a Debian-based Docker image
+([`pyodide/pyodide-env`](https://hub.docker.com/r/pyodide/pyodide-env)) on
+Docker Hub with the dependencies already installed to make it easier to build
+Pyodide. On top of that we provide
+a pre-built image
+([`pyodide/pyodide`](https://hub.docker.com/r/pyodide/pyodide)) which can be
+used for fast custom and partial builds. Note that building from the non
+pre-built the Docker image is _very_ slow on Mac, building on the host machine
+is preferred if at all possible.
+
+1. Install Docker
+
+1. From a git checkout of Pyodide, run `./run_docker` or `./run_docker --pre-built`
+
+1. Run `make` to build.
+
+```{note}
+You can control the resources allocated to the build by setting the env
+vars `EMSDK_NUM_CORE`, `EMCC_CORES` and `PYODIDE_JOBS` (the default for each is
+4).
+```
+
+If running `make` deterministically stops at one point,
+increasing the maximum RAM usage available to the docker container might help
+[This is different from the physical RAM capacity inside the system]. Ideally,
+at least 3 GB of RAM should be available to the docker container to build
+Pyodide smoothly. These settings can be changed via Docker Preferences (See
+[here](https://stackoverflow.com/questions/44533319/how-to-assign-more-memory-to-docker-container)).
+
+You can edit the files in your source checkout on your host machine, and then
```

Comment:
> in your source checkout

Slightly weird wording -- maybe "in the local copy of the source" or something?

Reply:
I guess the point here is that the source folder is shared between the docker image and the host machine. The docker image doesn't even have vim or any sort of text editor, so for editing the files in there the only options are ed or sed...

Reply:
Indeed. Tried to improve the wording.